packages/discord.js/src/structures/MessageMentions.js

```diff
@@ -185,11 +190,14 @@ class MessageMentions {
   get channels() {
     if (this._channels) return this._channels;
     this._channels = new Collection();
+    const globalChannelsPatern = new RegExp(this.constructor.ChannelsPattern.source, 'g');
     let matches;
-    while ((matches = this.constructor.ChannelsPattern.exec(this._content)) !== null) {
-      const chan = this.client.channels.cache.get(matches[1]);
-      if (chan) this._channels.set(chan.id, chan);
+
+    while ((matches = globalChannelsPatern.exec(this._content)) !== null) {
```

Comment:
Can't this just use matchAll and a for-of?

Reply:
Discussed internally, this pattern yields 2x the performance compared to `string.matchAll(regexp)`.